v2/contribute/decisions/session/0022-ssr-jwt-use-case.mdx

```diff
@@ -0,0 +1,63 @@
+---
+id: "0022"
+title: Introduce a new setting to expose entire access token to FE
+hide_title: true
+---
+
+import DecisionHeader from "/src/components/decisionLogs/DecisionHeader"
+import ArgumentPro from "/src/components/decisionLogs/ArgumentPro"
+import ArgumentCon from "/src/components/decisionLogs/ArgumentCon"
+import ArgumentNeut from "/src/components/decisionLogs/ArgumentNeut"
+
+# Introduce a new setting to expose entire access token to FE
+
+<DecisionHeader status="proposed" lastUpdate="2022-12-06" created="2022-12-06" deciders={["rishabhpoddar", "porcellus"]} proposedBy={["porcellus"]} />
+
+## Context and Problem Statement
+
+In the SSR + JWT (i.e.: websocket) usecase, the user may want to access the entire access token on the FE, but also needs the BE to access them during SSR. To access them on the FE we'd have to use header based auth, but that'd mean that the BE wouldn't have access to it during SSR.
+
+## Considered Options
+
+* Use header-based auth and set cookies for auth on the FE
+* Always set cookies (even in header mode)
+* Introduce a new auth mode
+* Introduce a new setting (`exposeAccessTokenToFrontendInCookieBasedAuth`)
+
```

Comment:
> Changes to the access token would cause issues even without an exposed access token: we have no way to update the front-token so payload updates would be lost/desynchronised with the FE.

Which is why i said that we should maybe make an API which the frontend can call in case it wants the access token. In this case, the API would also always cause an update to the front-token.

> I think we should add something like a forSSR setting to getSession.

I'm open to this, but this is very easy to miss. I think we can just ignore this issue alltogether. 

> "provide the session object in the shape that is expected by the respective prop of session auth"

I don't think that the backend getSession should care about the shape of that prop since SessionAuth is a frontend, react only thing.

Reply:
The upside of adding an SSR version of the session container that matches what we expect in react is that people could use that directly to initialize `SessionAuth` and do proper SSR on things that require sessions. Right now that's not possible, the inside we do not render until the session is loaded in the browser if it requires a session.

Reply:
Leaving this open cause of this https://github.com/supertokens/supertokens-node/issues/469